packages/core/src/grpc/decorators/service.decorator.ts

```diff
@@ -0,0 +1,13 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+// tslint:disable:no-any
+
+// TODO(bajtos) serviceSpec needs to be typed
+export function service(serviceSpec: any) {
+  return function(controllerCtor: Function) {
```

Comment:
Is it important that we append the property to the target as-is, or can we extend it with the decorator?
```
export function service(serviceSpec: any) {
  return function newClass<T extends {new(...args: any[]):{}}>(target: T) {
      return class extends target {
        grpcService: any = serviceSpec;
      }   
  }
};
```

Reply:
Good point. I think it should be ok to return a new class in the decorator. But then what are the downsides of appending the static property the existing controller class? Is the extra complexity worth it?

One of my concerns is that controllers decorated with `@services` will have anonymous class name or the same class name `newClass` for all controllers. This will make (error) stack traces rather unhelpful, as we are already experiencing with juggler models that all have the sam class name `ModelConstructor` (see [lib/model-builder.js](https://github.com/strongloop/loopback-datasource-juggler/blob/6d4cb6c3ade00b02c3d84997668697aa3959e0ed/lib/model-builder.js#L189-L197)).